source/clear-linux/guides/maintenance/cpu-performance.rst

```diff
@@ -0,0 +1,176 @@
+.. _cpu-performance:
+
+CPU Power and Performance
+#########################
+
+Modern x86 CPUs employ a number of features and technologies to balance
+performance, energy, and thermal efficiencies.
+
+By default, |CL| prioritizes maximum CPU performance with the philosophy that
+the faster programs finishes execution, the faster the CPU can return to a low
+energy idle state. It is important to understand and evaluate all of these
+technologies when troubleshooting or considering changing the defaults.
+
+
+.. contents:: :local:
+    :depth: 2
+
+
+CPU clock frequency scaling
+===========================
+
+The CPUFreq subsystem in Linux allows the OS to control C-states and P-states
+via CPU drivers and governors that provide algorithms defining how and when to
+enter these states. 
+
+
+Scaling driver
+--------------
+
+Linux uses the `Intel P-state driver`_, *intel_pstate*, for modern Intel
+processors from the Sandy Bridge generation or newer. Other processors may
+default to the *acpi-cpufreq* driver which reads values from the systems UEFI
+or BIOS.
+
+To view the current CPU frequency scaling driver run this command in a terminal:
+
+.. code:: bash
+
+   cat /sys/devices/system/cpu/cpu*/cpufreq/scaling_driver
+
+
+Scaling governor
+----------------
+
+|CL| sets the CPU governor to *performance* which calls for the CPU to operate
+at maximum clock frequency. In other words, P-state P0. While this may sound
+wasteful at first, it is important to remember that power utilization does not
+increase significantly simply because of a locked clock frequency without a
+workload.
+
+To view the current CPU frequency scaling governor run this command in a terminal:
+
+.. code:: bash
+   
+   cat /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor
+
+
+To change the CPU frequency scaling governor:
+
+#. Disable |CL| enforcement of certain power and performance settings.
+
+   .. code:: bash
+
+      sudo systemctl mask clr-power.timer
+
+
+#. Change the governor. In the example below, the governor is set to
+   *performance*.
+
+   .. code:: bash
+
+      echo performance | sudo tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor 
+   
+
+The list of all governors can be found in the `Linux kernel documentation on
+CPUFreq Governors`_.
+
+.. note::
+
+  The intel_pstate driver only supports *performance* and *powersave* governors.
+
+
+Thermal management
+==================
+
+`thermald`_ is a Linux thermal management daemon used to prevent the
+overheating of platforms. It monitors temperature and controls available
+cooling methods accordingly. This can be especially desirable for laptops.
```

Comment:
doesn't explain how it does this - specifically, it inserts sleep cycles (forces the cpu into a C-state).